sw/device/silicon_creator/rom/e2e/rom_e2e_c_init_test.c

```diff
@@ -0,0 +1,184 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/abs_mmio.h"
+#include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/csr.h"
+#include "sw/device/lib/base/hardened.h"
+#include "sw/device/silicon_creator/lib/drivers/otp.h"
+#include "sw/device/silicon_creator/lib/manifest_def.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "otp_ctrl_regs.h"
+#include "pinmux_regs.h"
+#include "uart_regs.h"
+
+enum {
+  /**
+   * Base address of the pinmux registers.
+   */
+  kPinmuxBase = TOP_EARLGREY_PINMUX_AON_BASE_ADDR,
+  /**
+   * Base address of the UART registers.
+   */
+  kUartBase = TOP_EARLGREY_UART0_BASE_ADDR,
+};
+
+/**
+ * Triggers an instruction access fault.
+ */
+static void fault(void) {
+  ((void (*)(void))TOP_EARLGREY_SRAM_CTRL_MAIN_RAM_BASE_ADDR)();
+}
+
+/**
+ * Triggers an instruction access fault if the condition is not true.
+ */
+#define CHECK(condition) \
+  do {                   \
+    if (!(condition)) {  \
+      fault();           \
+    }                    \
+  } while (false)
```

Comment:
Can you add a `LOG_ERROR()` (preferably with a message) so that we can debug failures easily.